### ISSUE TITLE
Can't build rpm for CentOS 8 using Docker

### ISSUE THREAD

**Issue by karljohns0n** [rpm-software-management/mock#447]
### Short description of the problem

I use CentOS 7 docker image to test my builds for CentOS 6/7/8 and mock fail to build el8 packages (using docker only):

```
INFO: Mock Version: 1.4.21
Start: dnf update
Finish: dnf update
Finish: chroot init
Start: build phase for aeris-release-1.0-6.el8.src.rpm
Start: build setup for aeris-release-1.0-6.el8.src.rpm
Finish: build setup for aeris-release-1.0-6.el8.src.rpm
Start: Outputting list of installed packages
ERROR: Exception(/tmp/aeris-release/SRPMS/aeris-release-1.0-6.el8.src.rpm) Config(aeris-8-x86_64) 0 minutes 52 seconds
INFO: Results and/or logs in: /tmp/aeris-release/RPMS
ERROR: state finish mismatch: current: Outputting list of installed packages, state: build phase for aeris-release-1.0-6.el8.src.rpm

The only errors in root.log:

DEBUG util.py:689:  Executing command: rpm -qa --root '/var/lib/mock/aeris-8-x86_64/root' --qf '%{nevra} %{buildtime} %{size} %{pkgid} installed\n' > /tmp/aeris-release/RPMS/installed_pkgs.log with env {'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOME': '/builddir', 'HOSTNAME': 'mock', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'LANG': 'C.UTF-8'} and shell True
DEBUG util.py:689:  Executing command: rpm -qa --root '/var/lib/mock/aeris-8-x86_64/root' --qf '%{nevra} %{buildtime} %{size} %{pkgid} installed\n' > /tmp/aeris-release/RPMS/installed_pkgs.log with env {'TERM': 'vt100', 'SHELL': '/bin/bash', 'HOME': '/builddir', 'HOSTNAME': 'mock', 'PATH': '/usr/bin:/bin:/usr/sbin:/sbin', 'PROMPT_COMMAND': 'printf "\\033]0;<mock-chroot>\\007"', 'PS1': '<mock-chroot> \\s-\\v\\$ ', 'LANG': 'C.UTF-8'} and shell True
DEBUG util.py:596:  error: db5 error(5) from dbenv->open: Input/output error
DEBUG util.py:596:  error: db5 error(5) from dbenv->open: Input/output error
DEBUG util.py:596:  error: cannot open Packages index using db5 - Input/output error (5)
DEBUG util.py:596:  error: cannot open Packages index using db5 - Input/output error (5)
DEBUG util.py:596:  error: cannot open Packages database in /var/lib/mock/aeris-8-x86_64/root/var/lib/rpm
DEBUG util.py:596:  error: cannot open Packages database in /var/lib/mock/aeris-8-x86_64/root/var/lib/rpm
DEBUG util.py:744:  Child return code was: 1
DEBUG util.py:744:  Child return code was: 1
DEBUG util.py:329:  kill orphans
DEBUG util.py:329:  kill orphans
``` 

Using mock with the same config on other hypervisors (openvz or kvm) will build without issue. Mock also build CentOS 6 and 7 packages using docker without issue. This looks like #367.

### Output of `rpm -q mock`

```
mock-1.4.21-1.el7.noarch
```
### Steps to reproduce issue

```
[root@xenpv -]# docker run --detach --privileged -v /sys/fs/cgroup:/sys/fs/cgroup:ro --volume="${PWD}":/tmp/"${pkg_name}" karljohns0n/rpmbuild:7
[root@xenpv -]# docker exec $ID mock --old-chroot --clean -D "dist .el${version}" -r aeris-"${version}"-x86_64 --resultdir=/tmp/"${pkg_name}"/RPMS --rebuild /tmp/"${pkg_name}"/SRPMS/"${pkg_name}"-"${pkg_version}"-"${pkg_release}".el"${version}".src.rpm
```

### Any additional notes

```
[root@xenpv -]# docker exec $ID cat /etc/mock/aeris-8-x86_64.cfg
config_opts['root'] = 'aeris-8-x86_64'
config_opts['target_arch'] = 'x86_64'
config_opts['legal_host_arches'] = ('x86_64',)
config_opts['chroot_setup_cmd'] = 'install tar gcc-c++ redhat-rpm-config redhat-release which xz sed make bzip2 gzip gcc coreutils unzip shadow-utils diffutils cpio bash gawk rpm-build info patch util-linux findutils grep'
config_opts['dist'] = 'el8'
config_opts['releasever'] = '8'
config_opts['macros']['%dist'] = ".el8"
config_opts['package_manager'] = 'dnf'
config_opts['use_bootstrap_container'] = True
config_opts['bootstrap_image'] = 'centos:8'
config_opts['extra_chroot_dirs'] = [ '/run/lock', ]
config_opts['yum.conf'] = """

[main]
keepcache=1
debuglevel=2
reposdir=/dev/null
logfile=/var/log/yum.log
retries=20
obsoletes=1
gpgcheck=0
assumeyes=1
syslog_ident=mock
syslog_device=
mdpolicy=group:primary
best=1
protected_packages=
module_platform_id=platform:el8

[BaseOS]
....
```

```
[root@xenpv -]# docker exec $ID ls -al /var/lib/mock/aeris-8-x86_64/root/var/lib/rpm
total 13752
drwxr-xr-x.  2 root root     4096 Jan 12 23:28 .
drwxr-xr-x. 17 root root     4096 Jan 12 23:04 ..
-rw-r--r--.  1 root mock        0 Jan 12 23:00 .dbenv.lock
-rw-r--r--.  1 root mock        0 Jan 12 23:00 .rpm.lock
-rw-r--r--.  1 root mock  1081344 Jan 12 23:05 Basenames
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Conflictname
-rw-r--r--.  1 root mock   266240 Jan 12 23:05 Dirnames
-rw-r--r--.  1 root mock     8192 Jan 12 23:00 Enhancename
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Filetriggername
-rw-r--r--.  1 root mock     8192 Jan 12 23:05 Group
-rw-r--r--.  1 root mock    12288 Jan 12 23:05 Installtid
-rw-r--r--.  1 root mock    16384 Jan 12 23:05 Name
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Obsoletename
-rw-r--r--.  1 root mock 10723328 Jan 12 23:05 Packages
-rw-r--r--.  1 root mock    90112 Jan 12 23:05 Providename
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Recommendname
-rw-r--r--.  1 root mock    98304 Jan 12 23:05 Requirename
-rw-r--r--.  1 root mock    24576 Jan 12 23:05 Sha1header
-rw-r--r--.  1 root mock    24576 Jan 12 23:05 Sigmd5
-rw-r--r--.  1 root mock     8192 Jan 12 23:00 Suggestname
-rw-r--r--.  1 root mock     8192 Jan 12 23:00 Supplementname
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Transfiletriggername
-rw-r--r--.  1 root mock     8192 Jan 12 23:04 Triggername
-rw-r--r--.  1 root root   376832 Jan 12 23:28 __db.001
-rw-r--r--.  1 root root    98304 Jan 12 23:28 __db.002
-rw-r--r--.  1 root root  1318912 Jan 12 23:28 __db.003
```

**Comment by xsuchy** [comment]
Does this happened after an upgrade? Can you try `mock -r aeris-8-x86_64 --scrub=all` and try again?

**Comment by karljohns0n** [comment]
I use a fresh docker image for every build so the cache is empty. Same error after scrubbing:

```
[root@xenpv pkg-aeris-release]# docker exec f61128c41475 mock -r aeris-8-x86_64 --scrub=all
INFO: mock.py version 1.4.21 starting (python version = 3.6.8)...
Start(bootstrap): init plugins
INFO: selinux disabled
Finish(bootstrap): init plugins
Start: init plugins
INFO: selinux disabled
Finish: init plugins
INFO: Signal handler active
Start: run
Start: scrub ['all']
INFO: scrubbing everything for aeris-8-x86_64
Finish: scrub ['all']
Finish: run
[root@xenpv pkg-aeris-release]#
[root@xenpv pkg-aeris-release]# docker exec f61128c41475 mock --old-chroot --clean -D "dist .el${version}" -r aeris-"${version}"-x86_64 --resultdir=/tmp/"${pkg_name}"/RPMS --rebuild /tmp/"${pkg_name}"/SRPMS/"${pkg_name}"-"${pkg_version}"-"${pkg_release}".el"${version}".src.rpm
INFO: mock.py version 1.4.21 starting (python version = 3.6.8)...
Start(bootstrap): init plugins
INFO: selinux disabled
Finish(bootstrap): init plugins
Start: init plugins
INFO: selinux disabled
Finish: init plugins
INFO: Signal handler active
Start: run
INFO: Start(/tmp/aeris-release/SRPMS/aeris-release-1.0-6.el8.src.rpm)  Config(aeris-8-x86_64)
Start: clean chroot
Finish: clean chroot
Start(bootstrap): chroot init
INFO: calling preinit hooks
INFO: enabled root cache
INFO: enabled dnf cache
Start(bootstrap): cleaning dnf metadata
Finish(bootstrap): cleaning dnf metadata
INFO: enabled HW Info plugin
Mock Version: 1.4.21
INFO: Mock Version: 1.4.21
Start(bootstrap): yum install
Finish(bootstrap): yum install
Start(bootstrap): creating root cache
Finish(bootstrap): creating root cache
Finish(bootstrap): chroot init
Start: chroot init
INFO: calling preinit hooks
INFO: enabled root cache
INFO: enabled dnf cache
Start: cleaning dnf metadata
Finish: cleaning dnf metadata
INFO: enabled HW Info plugin
Mock Version: 1.4.21
INFO: Mock Version: 1.4.21
Start: dnf install
Finish: dnf install
Start: creating root cache
Finish: creating root cache
Finish: chroot init
INFO: Installed packages:
Start: build phase for aeris-release-1.0-6.el8.src.rpm
Start: build setup for aeris-release-1.0-6.el8.src.rpm
Finish: build setup for aeris-release-1.0-6.el8.src.rpm
Start: Outputting list of installed packages
ERROR: Exception(/tmp/aeris-release/SRPMS/aeris-release-1.0-6.el8.src.rpm) Config(aeris-8-x86_64) 9 minutes 42 seconds
INFO: Results and/or logs in: /tmp/aeris-release/RPMS
INFO: Cleaning up build root ('cleanup_on_failure=True')
Start: clean chroot
Finish: clean chroot
ERROR: state finish mismatch: current: Outputting list of installed packages, state: build phase for aeris-release-1.0-6.el8.src.rpm
[root@xenpv pkg-aeris-release]#
```

**Comment by karljohns0n** [comment]
You can see an example of the build on Travis CI [here](https://travis-ci.org/karljohns0n/pkg-aeris-release/jobs/636422535) for CentOS 6 and 7. I haven't enabled CentOS 8 on Travis yet as I can't build with docker (only docker) for now.

**Comment by karljohns0n** [comment]
Update: If I use mock with centos:8 image instead of centos:7 image, with the same mock config, it works.

**Comment by praiskup** [comment]
Your build seem to fail on:
> Start: Outputting list of installed packages

This should be fixed by 60b2b34f8f325d0b90dbfc7d0d294a4d1edac155.

It's not yet released, the fix will be in v2.2.
